django-rgd/rgd/permissions.py

```diff
@@ -87,22 +137,26 @@ def filter_perm(user, queryset, role):
     if user.is_active and user.is_superuser:
         return queryset
     # Check permissions
-    subquery = queryset.none()
-    # Now grab resources created by this user - only file-based models
-    if not issubclass(queryset.model, (models.CollectionPermission, models.Collection)):
-        for path in get_permissions_paths(queryset.model, models.ChecksumFile):
-            created_by_path = (path + '__' if path != '' else path) + 'created_by'
-            condition = Q(**{created_by_path: user})
-            subquery = subquery.union(queryset.filter(condition).values('pk'))
-    for path in get_permissions_paths(queryset.model, models.CollectionPermission):
-        # `path` can be an empty string (meaning queryset is `CollectionPermission`)
-        user_path = (path + '__' if path != '' else path) + 'user'
-        role_path = (path + '__' if path != '' else path) + 'role'
-        condition = Q(**{user_path: user}) & Q(**{role_path + '__gte': role})
-        if getattr(settings, 'RGD_GLOBAL_READ_ACCESS', False):
-            condition |= Q(**{path + '__isnull': True})
-        subquery = subquery.union(queryset.filter(condition).values('pk'))
-    return queryset.filter(pk__in=subquery)
+    conditions = Q()
+    model = queryset.model
+    for path in get_paths(model, models.ChecksumFile):
+        # A user can read/write a file if they are the creator
+        is_creator = path.q(created_by=user)
+        conditions |= is_creator
+        if (
+            getattr(settings, 'RGD_GLOBAL_READ_ACCESS', False)
+            and role == models.CollectionPermission.READER
+        ):
+            # A user can read a file by default if it is not in a collection
+            is_not_in_collection = path.q(collection__isnull=True)
+            has_no_owner = path.q(created_by__isnull=True)
+            conditions |= is_not_in_collection & has_no_owner
```

Comment:
See comment on this authorization logic in #608. The commit over there, https://github.com/ResonantGeoData/ResonantGeoData/pull/608/commits/a8bce31040866b5f7d441112f38efb698c85421c, changes this behavior.